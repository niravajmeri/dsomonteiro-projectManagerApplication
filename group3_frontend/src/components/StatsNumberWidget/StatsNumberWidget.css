.number-widget {
    background-color: #ffffff;
    margin: 5px;
    flex-grow: 1;
    flex-basis: 30%;
    max-width: 50%;
}

.number-widget > .widget-title {
    font-size: 20px;
    margin: 10px;
    font-weight: bold;
}

.number-widget .widget-values .widget-unit {
    font-size: 15px;
    margin-left: 5px;
}

.number-widget .widget-values {
    text-align: center;
    line-height: 60px;
}
.number-widget .widget-values .widget-value {
    font-size: 60px;
}
