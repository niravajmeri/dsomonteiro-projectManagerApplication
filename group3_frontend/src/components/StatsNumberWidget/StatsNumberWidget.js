import React, { Component } from 'react';
import './StatsNumberWidget.css';

class StatsNumberWidget extends Component {
    state = {};
    render() {
        return (
            <div className="number-widget">
                <h1 className="widget-title">{this.props.title}</h1>
                <div className="widget-values">
                    <span className="widget-value">{this.props.value}</span>
                    <span className="widget-unit">{this.props.unit}</span>
                </div>
            </div>
        );
    }
}

export default StatsNumberWidget;
