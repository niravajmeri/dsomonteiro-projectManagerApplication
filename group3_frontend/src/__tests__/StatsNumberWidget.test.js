import React from 'react';
import { shallow } from 'enzyme';
import StatsNumberWidget from '../components/StatsNumberWidget/StatsNumberWidget';

describe('StatsNumberWidget', () => {
    it('should render correctly', () => {
        const component = shallow(
            <StatsNumberWidget value="1" title="title" unit="unit" />
        );
        expect(component).toMatchSnapshot();
    });
});
