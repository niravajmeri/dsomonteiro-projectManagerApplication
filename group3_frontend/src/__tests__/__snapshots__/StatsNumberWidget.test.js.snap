// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsNumberWidget should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StatsNumberWidget
    title="title"
    unit="unit"
    value="1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="widget-title"
>
          title
</h1>,
        <div
          className="widget-values"
>
          <span
                    className="widget-value"
          >
                    1
          </span>
          <span
                    className="widget-unit"
          >
                    unit
          </span>
</div>,
      ],
      "className": "number-widget",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "title",
          "className": "widget-title",
        },
        "ref": null,
        "rendered": "title",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="widget-value"
>
              1
</span>,
            <span
              className="widget-unit"
>
              unit
</span>,
          ],
          "className": "widget-values",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1",
              "className": "widget-value",
            },
            "ref": null,
            "rendered": "1",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "unit",
              "className": "widget-unit",
            },
            "ref": null,
            "rendered": "unit",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="widget-title"
>
            title
</h1>,
          <div
            className="widget-values"
>
            <span
                        className="widget-value"
            >
                        1
            </span>
            <span
                        className="widget-unit"
            >
                        unit
            </span>
</div>,
        ],
        "className": "number-widget",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
            "className": "widget-title",
          },
          "ref": null,
          "rendered": "title",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="widget-value"
>
                1
</span>,
              <span
                className="widget-unit"
>
                unit
</span>,
            ],
            "className": "widget-values",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1",
                "className": "widget-value",
              },
              "ref": null,
              "rendered": "1",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "unit",
                "className": "widget-unit",
              },
              "ref": null,
              "rendered": "unit",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
